### PR TITLE
TS-4377: update documentation links in default configuration files

### PR DESCRIPTION
**Pull request by jsime**
- Updates existing links in default configuration files to point to current Administrator's Guide documentation URLs.
- Adds links to TS documentation for all default configuration files which a) did not already have them, and b) which have documentation.

### CODE REVIEW

File: proxy/config/records.config.default.in [hunk 1]
```diff
@@ -15,12 +15,11 @@ CONFIG proxy.config.ssl.number.threads INT -1
 CONFIG proxy.config.accept_threads INT 1
 CONFIG proxy.config.task_threads INT 2
 CONFIG proxy.config.cache.threads_per_disk INT 8
-    # https://docs.trafficserver.apache.org/records.config#proxy-config-exec-thread-affinity
 CONFIG proxy.config.exec_thread.affinity INT 1
 
 ##############################################################################
 # Specify server addresses and ports to bind for HTTP and HTTPS. Docs:
-#    https://docs.trafficserver.apache.org/records.config#proxy-config-http-server-ports
+#  i  https://docs.trafficserver.apache.org/records.config#proxy.config.http.server_ports
```
Comment (maskit) [inline proxy/config/records.config.default.in]: There is a superfluous "i" at the beginning.

**Comment by maskit** [comment]
Some URLs are shorten. They should be unified?

e.g.
https://docs.trafficserver.apache.org/records.config#thread-variables
https://docs.trafficserver.apache.org/en/latest/admin-guide/files/records.config.en.html

**Comment by jsime** [comment]
Thanks for the catches. Fixed that 'i' typo and I switched the full URL in the comment at the top of records.config.default.in over to the shortened version. The links to other config files' documentation pages still reference the full URLs because we don't have redirects set up for those on the docs host right now. That could be a useful thing to have, though.

**Comment by maskit** [comment]
Looks good to me.

**Comment by jpeach** [comment]
@jsime Ready to merge this one?

**Pull request merged**